spec/ics-2-consensus-verification/README.md

```diff
@@ -0,0 +1,324 @@
+level protocol abstractions. The algorithm which uses these properties to verify substates of
+another chain is referred to as a "light client verifier", and pairing it with a state that the
+verifier trusts forms a "light client".
+
+This standard also specifies how the light clients will be stored, registered, and updated on a
+blockchain. The stored light client instances will be able to be verified by a third party actor, such as a user inspecting the state of the chain.
+
+## Specification
+
+### Motivation
+
+In the IBC protocol, a chain needs to be able to verify updates to the state of another chain. A light client is the algorithm with which they can do so.
+updating state with a light client pointing to that chain. This standard formalises the common
+model of light client to minimise the dependency on consensus algorithms, so that the protocol can
+easily connect with new chains which are running new consensus algorithms, without need to
+upgrade the light client protocol itself.
+
+<!--
+`FullNode`s are procedures running a `Consensus`. Given a `([Transaction], Commit)`, a
+`FullNode` can compute the result `ConsensusState` that the `Consensus` is expected to commit on
+with the same `[Transaction]`, if exists.
+--->
+### Desired Properties
+
+This standard specification provides secure layer to verify other chains' canonical headers,
+using the existing `ConsensusState`. The higher level abstractions will then be able to verify subcomponents of the state
+with the `CommitmentRoot` stored in the `ConsensusState`, which is guaranteed to be committed by
+the other chain's consensus algorithm.
+
+* `ValidityPredicate`s are expected to reflect the behaviour of the full node which is running the  
+corresponding consensus algorithm. Given a `ConsensusState` and `[Message]`, if a full node
+accepts the new `Header` generated with `Commit`, then the light client MUST also accept it,
+and if a full node rejects it, then the light client MUST also reject it. The consensus algorithm
+ensures this correspondence. However light clients are not replaying the whole messages, so it
+is possible that the light clients' behaviour differs from the full nodes'. In this case, the
+equivocation proof which proves the divergence between the `ValidityPredicate` and the full node will be
+generated and submitted to the chain, as defined in
+[ICS ?](https://github.com/cosmos/ics/issues/53), so that the chain can safely deactivate the
+light client.
+
+### Technical Specification
+
+#### Requirements
+
+* `get`, `set`, `Key`, and `Identifier` are as defined in [ICS24](../ics-24-host-requirements).
+are used by the datagram handler.
+
+* `CommitmentRoot` is as defined in [ICS23](https://github.com/cosmos/ics/pull/74).
+`ConsensusState`. The downstream logic can use it to verify whether key-value pairs are present
+in the state or not.
+
+* `createClient`, `queryClient`, `updateClient`, `freezeClient`, `deleteClient` are as
```

Comment:
Well, the function *definitions* are defined here - the function signatures are defined in ICS 25, yes